sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/BenchmarkHelper.java

```diff
@@ -28,15 +28,10 @@ public static boolean shouldContinue(long startTimeMillis, long iterationCount,
         if (maxDurationTime == null) {
             return iterationCount < maxNumberOfOperations;
         }
-
+        
         if (startTimeMillis + maxDurationTime.toMillis() < System.currentTimeMillis()) {
             return false;
         }
-
-        if (maxNumberOfOperations < 0) {
-            return true;
-        }
-
-        return iterationCount < maxNumberOfOperations;
+        return true;
```

Comment:
I am not sure how this fixes the issue?
To fix the `maxDurationTime` issue, shouldn't we change the above if condition?

Reply:
Case 1: maxDurationTime is null
1st if condition is activated and maxNumberofOperations property is used

Case 2: maxDurationTime is populated for example 48hrs
2nd if condition will be kicked in when the startTIme of the workload + 48hrs is greater than the current time, meaning 48hrs are over. Until then it would return true, so the workload will keep on running. 

This change was tested in the AsyncNonBlocking Cache testing workloads. I just missed pushing it with that PR.

Reply:
Exactly, here the condition is not greater than, rather less than -> 

`startTimeMillis + maxDurationTime.toMillis() < System.currentTimeMillis()`

Also, what about if both maxDurationTime and maxNumberOfOperations are passed. In that case, earlier the code was stopping on condition whichever reached first, but now we are not doing that.

Reply:
You're right it should be greater than instead of less than. I'll update the if statement.

Also, both the maxDurationTime and maxNumberOfOperations properties are used to set the duration of the workload. It doesn't make any sense to use both properties together.

But I can add a check for maxNumberOfOperations, if that is required.

Reply:
That's okay, if both are not supposed to be used together, we should be fine.